tests/benchmarks/test_xgboost.py

```diff
@@ -0,0 +1,237 @@
+        "DOBorough",
+        "PUSuperborough_DOSuperborough",
+    ]
+
+    # TODO https://github.com/dask/dask/issues/9847
+    ddf = ddf.astype(dict.fromkeys(categories, "category"))
+    ddf = ddf.persist()
+    # This blocks until the whole workload so far has been persisted
+    ddf = ddf.categorize(categories)
+
+    ########
+    # Output
+    ########
+    ddf = ddf.persist().repartition(partition_size="100MB")
+    # At this point we would normally finish with to_parquet()
+    wait(ddf, small_client, timeout=600)
+
+
+@run_up_to_nthreads("small_cluster", 200, reason="fixed size dataset")
+def test_optuna_hpo(small_client):
+    # Needs either coiled-runtime >0.2.1 or explicitly adding the package to the A/B
+    # test environment
+    optuna = pytest.importorskip("optuna")
+
+    #############################
+    # Dataset load and preprocess
+    #############################
+    # Load feature table generated by previous test
+    ddf = dd.read_parquet(
+        "s3://coiled-runtime-ci/xgboost/nyc-uber-lyft-preprocessed.parquet",
+    )
+
+    # Full size dataset on 40x r6i.large
+    ddf = downscale_dataframe(ddf, "580 GiB")
+
+    # Under the hood, XGBoost converts floats to `float32`.
+    # Let's do it only once here.
+    float_cols = ddf.select_dtypes(include="float").columns.tolist()
+    ddf = ddf.astype({c: np.float32 for c in float_cols})
+
+    # We need the categories to be known
+    categorical_vars = ddf.select_dtypes(include="category").columns.tolist()
+
+    # categorize() reads the whole input and then discards it.
+    # Let's read from disk only once.
+    ddf = ddf.persist()
+    ddf = ddf.categorize(columns=categorical_vars)
+
+    ####################################
+    # Split between train and test data
+    # and initialize the XGBoost DMatrix
+    ####################################
+    train, test = ddf.random_split([0.8, 0.2], shuffle=True)
+    x_train = train.drop(columns=["trip_time"])
+    y_train = train["trip_time"]
+    x_test = test.drop(columns="trip_time")
+    y_test = test["trip_time"]
+
+    # We will need to access these multiple times. Let's persist them.
+    x_test, y_test = small_client.persist([x_test, y_test])
+
+    # Release no longer necessary objects on the cluster
+    del ddf, train, test
+
+    # Build XGBoost matrix. This is automatically persisted.
+    d_train = xgboost.dask.DaskDMatrix(None, x_train, y_train, enable_categorical=True)
+
+    del x_train, y_train
+
+    #########################################
+    # Have Optuna call xgboost.train multiple
+    # times with different hyperparameters
+    #########################################
+    def train_model(**study_params):
+        model = xgboost.dask.train(
+            None,
+            {"tree_method": "hist", **study_params},
+            d_train,
+            num_boost_round=4,
+            evals=[(d_train, "train")],
+        )
+        predictions = xgboost.dask.predict(None, model, x_test)
```

Comment:
by looking at XGBoost examples we probably should we put a comment explaining why we are not generating the DMatrix for test. (btw I do not know the reason)

Reply:
> the xgboost.dask examples suggests an unnecessary and expensive intermediate step. 

Thanks for clarifying this, I was not aware that step was unnecessary. I assumed their docs were following something standard, but I guess they are not.